restart_plugin/mtcp_restart_plugin.c

```diff
@@ -849,6 +849,12 @@ mtcp_plugin_skip_memory_region_munmap(Area *area, RestoreInfo *rinfo)
   LhCoreRegions_t *lh_regions_list = NULL;
   int total_lh_regions = lh_info->numCoreRegions;
 
+  // Don't skip munmap of mtcp_restart regions.
+  if (mtcp_strendswith(area->name, "/mtcp_restart") ||
+      mtcp_strendswith(area->name, "[heap]")) {
```

Comment:
Update: For some reason unmap the heap region here would cause seg fault, but without unmapping it we will encounter conflict as well later. @karya0